librz/core/cbin.c

```diff
@@ -1622,6 +1622,46 @@ RZ_API bool rz_core_bin_apply_resources(RzCore *core, RzBinFile *binfile) {
 	return true;
 }
 
+static void digests_ht_free(HtPPKv *kv) {
+	free(kv->key);
+	free(kv->value);
+}
+
+/**
+ * \brief Create a hashtable of digests for the RzBinSection
+ *
+ * Digest names are supplied as a list of `char *` strings.
+ * Returns the hashtable with keys of digest names and values of
+ * strings containing requested digests.
+ * */
+RZ_API RZ_OWN HtPP *rz_core_bin_section_digests(RzCore *core, RzBinSection *section, RzList *digests) {
+	rz_return_val_if_fail(section && digests, NULL);
+	HtPP *r = ht_pp_new(NULL, digests_ht_free, NULL);
+	if (!r) {
+		goto err;
+	}
+	RzListIter *it;
+	char *digest;
+	rz_list_foreach (digests, it, digest) {
+		ut8 *data = malloc(section->size);
+		if (!data) {
+			ht_pp_free(r);
+			return NULL;
+			goto err;
+		}
+		ut32 datalen = section->size;
+		rz_io_pread_at(core->io, section->paddr, data, datalen);
+		char *chkstr = rz_msg_digest_calculate_small_block_string(digest, data, datalen, NULL, false);
```

Comment:
@wargio isn't there an API that takes a `RzBuffer` as input? I think that is a must nowadays, because we never want to have to allocate an whole buffer to compute digests.

Reply:
I checked for it in `librz/include/rz_msg_digest.h` and there is nothing.

Reply:
no, there is not.